app/modules/parsing/knowledge_graph/inference_service.py

```
@@ -465,23 +789,95 @@ async def generate_docstrings(self, repo_id: str) -> Dict[str, DocstringResponse
+                                    )
+
+                        # Handle chunk consolidation before Neo4j update
+                        processed_response = self.process_chunk_responses(
+                            response, batch
+                        )
+                        if processed_response:
+                            self.update_neo4j_with_docstrings(
+                                repo_id, processed_response
+                            )
+
+                    return response
```

Ruff (0.14.4)

Consider moving this statement to an `else` block

(TRY300)